app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -96,34 +139,106 @@ public void cleanup() {
 
         @Override
         public void setBlockingEnabled(boolean enabled) {
-            // We can't actually do this?
+            if (enabled) {
+                updateBlocking();
+            } else {
+                if (geckoSession != null) {
+                    geckoSession.disableTrackingProtection();
+                }
+            }
+            if (callback != null) {
+                callback.onBlockingStateChanged(enabled);
+            }
         }
 
-        private ContentListener createContentListener() {
-            return new ContentListener() {
+        @Override
+        public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String prefName) {
```

Comment:
\[PMD\] INFO: The method 'onSharedPreferenceChanged' has a Cyclomatic Complexity of 7\.